### ISSUE TITLE
Failed to prepare the private registry for both Windows and Linux images 

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#22836]
BUG

I run a private registry in a Ubuntu box, tag and push the image `rancher/rancher-agent:v2.3.0-rc1` to it, now I run a windows box and  pull the same image, then the error occurs
```
PS C:\Users\Administrator> docker pull my-registry/rancher/rancher-agent:v2.3.0-rc1
v2.3.0-rc1: Pulling from rancher/rancher-agent
image operating system "linux" cannot be used on this platform
```

Cause:

The current `rancher-load-images.sh`, `rancher-save-images.sh` and the documentation does not provide the method to prepare the private registry for both Windows and Linux.

**Comment by drpebcak** [comment]
This also needs to be fixed for `arm64`.

**Comment by cloudnautique** [comment]
Procedurally, I'm onboard, it should look something like this:

Input file:
`rancher/image:version amd64;remote/repo:version windows;remote/repo:version arm;remote/repo:version`

Ideally only the first field is positional. 

Windows script would be run first. 
rancher-load-images.ps1 and rancher-save-images.ps1
It would use the input file and only process items that `windows;` items. Is it reasonable to assume that the version string will include the OS specifics? if so, would we tag that with `rancher/image:version-1903` for instance?

On linux:
Load both linux and arm.
Rancher save would push.

Then a 3rd script for manifesting... or flags on the linux would run with the ability to --skip-windows, --skip-armhf, or --skip-skip-arm64. Armhf might not be needed. I don't think there is any harm in manifesting things that don't have multiple archs/OSes.

**Comment by drpebcak** [comment]
Yeah so the format that I think we agreed on was like this (for a somewhat concrete example, although I don't think etcd actually supports all of these versions):
`rancher/coreos-etcd:v1 amd64;coreos/etcd:v1 arm;coreos/etcd:v1-arm arm64;coreos/etcd:v1-arm64 windows-1903;coreos/etcd:v1-windows-1903`
and we will have the following tags in our org afterwards:
```
rancher/coreos-etcd:v1
rancher/coreos-etcd:v1-arm
rancher/coreos-etcd:v1-arm64
rancher/coreos-etcd:v1-windows-1903
```

So then from the user side, they can use this same file to pull these for each architecture (running from a computer that matches that arch).

If there is a windows entry on a line, pull the image like `rancher/coreos-etcd:v1-windows-1903`
If there is an erm entry on a line, pull the image like `rancher/coreos-etcd:v1-arm`
etc.

From there they just tag and push those images up to their private registry normally:
`docker tag rancher/coreos-etcd:v1-arm64 <custom_registry>/rancher/coreos-etcd:v1-arm64`
`docker push <custom_registry>/rancher/coreos-etcd:v1-arm64`

The final step will be creating a manifest for the images they have pushed to their registry. This manifest script can technically run from any arch.

**Comment by loganhz** [comment]
How can we find out the images list for arm? We have some images only support amd.

Maybe we need to pull all archs for each image. And check if it is supported in the arch?

**Comment by drpebcak** [comment]
@loganhz the exact same thing is true for windows. How do we currently detect if there is a windows version? This new file would be generated from something that lists the images/archs we support. If there is an arm version it will be in the list. If there isn't, it won't list one.

**Comment by loganhz** [comment]
We have some logic in code and charts. As arm is an experimental feature, so we don't support to list them now. So we can only list AMD and Windows for now. 

@deniseschannon should we support ARM in the scripts?

**Comment by sangeethah** [comment]
Instructions for pushing windows images - https://github.com/rancher/docs/issues/1742#issuecomment-532553582

**Comment by jiaqiluo** [comment]
The bug fix is validated with Rancher:v2.3.0-rc5 following the instruction mentioned in the last comment.
